src/Apps/Legion/Components/CookSource/CookSource.types.ts

```diff
@@ -0,0 +1,17 @@
+import { IStyle } from '@fluentui/react';
+import { IADXAdapterTargetContext } from '../../Models/Interfaces';
+import { SourceType } from '../DataPusher/DataPusher.types';
+import { ITable } from '../../Adapters/Standalone/DataManagement/Models/DataManagementAdapter.types';
+import { ICookSource } from '../../Models';
+
+export interface ICookSourceProps {
+    onSourceTypeChange: (sourceType: SourceType) => void;
+    onSourceChange: (source: ICookSource) => void;
+    onGetTableData?: (table: ITable) => void;
+    targetAdapterContext?: React.Context<IADXAdapterTargetContext>;
```

Comment:
how does context as a prop work? Why not useContext instead?

Reply:
So it is a context property that our useADXAdapter hook will try to fetch the adapter from. inside that hook I use useContext. If there is no context provided to that hook via targetAdapterContext, it will return MockAdapter